src/mobile-app/components/app.tsx

```diff
@@ -1,21 +1,90 @@
-import React from "react";
+import React, { useEffect } from "react";
 import { useState } from "react";
 import { ExperimentPicker } from "./experiment-picker";
 import { ExperimentWrapper } from "./experiment-wrapper";
-import { IExperiment } from "../../shared/experiment-types";
+import { IExperiment, IExperimentData } from "../../shared/experiment-types";
+import { LocalDataStorage } from "../../shared/utils/local-data-storage";
+import { RunPicker } from "./run-picker";
+import { IRun } from "./types";
+import { useExperiments } from "../hooks/use-experiments";
 
 import css from "./app.module.scss";
 
-export const AppComponent: React.FC<{}> = () => {
+const runsStorage = new LocalDataStorage<IRun>("runs-");
+const availableRunKeysStorage = new LocalDataStorage<string[]>("available-runs");
 
-  const [experiment, setExperiment] = useState<IExperiment|undefined>();
+export const AppComponent: React.FC = () => {
+  const { experiments, upgradeApp } = useExperiments();
+  const [runKey, setRunKey] = useState<string | undefined>();
```

Comment:
I think that all this code around saving/loading runs should be isolated into a single hook so that it can be tested outside the component and so the component can be focused on rendering the data.

Reply:
Right, I thought about it after I wrote that code. I'll try to move it to a hook.